roadmap.md

```diff
@@ -0,0 +1,54 @@
+# WebDriver BiDi roadmap
+
+## Real-world end-to-end user scenarios
+
+This document presents an overview of real-world end-to-end user scenarios we aim to enable via the WebDriver BiDi protocol. Each scenario requires one or more WebDriver BiDi commands and events to be specified, tested, and implemented across browser engines.
+
+### A. Extracting content
+
+This scenario loads a web page and uses script evaluation to extract content on the page (e.g. the headlines). In spec terms, this involves:
+
+- [x] [handling sessions](https://w3c.github.io/webdriver-bidi/#module-session)
+- [x] [creating a new `browsingContext`](https://w3c.github.io/webdriver-bidi/#command-browsingContext-create)
+- [x] [navigating to a URL](https://w3c.github.io/webdriver-bidi/#command-browsingContext-navigate)
+- [x] [evaluating JavaScript in the page context](https://w3c.github.io/webdriver-bidi/#command-script-evaluate)
+- [x] [serialization/deserialization of JavaScript values](https://w3c.github.io/webdriver-bidi/#data-types-protocolValue)
+- [ ] [sandboxed script execution](https://github.com/w3c/webdriver-bidi/issues/144)
+
+### B. Submitting forms
+
+This scenario loads a web page, enter text into a form field via the keyboard, and submits the form via a mouse click before extracting the results from the page. In spec terms, this involves:
+
+- [x] (everything from the previous milestone)
+- [ ] [emulating keyboard input](https://github.com/w3c/webdriver-bidi/pull/175)
+- [ ] [emulating mouse input](https://github.com/w3c/webdriver-bidi/pull/175)
+
+### C. Capturing screenshots
+
+This scenario loads a web page and captures a screenshot. In spec terms, this involves:
+
+- [x] (some items from the previous milestones)
+- [ ] [capturing a screenshot via a command](https://w3c.github.io/webdriver-bidi/#command-browsingContext-captureScreenshot)
+
+### D. Printing to PDF
+
+This scenario loads a web page and prints it as a PDF. In spec terms, this involves:
+
+- [x] (some items from the previous milestones)
+- [ ] [printing to PDF via a command](https://github.com/w3c/webdriver-bidi/issues/210)
+
+### E. Blocking images
+
+This scenario loads a web page and uses request interception to block any images from loading. In spec terms, this involves:
+
+- [x] (some items from the previous milestones)
+- [ ] [intercepting requests](https://github.com/w3c/webdriver-bidi/issues/66)
+
+### F. Measuring performance
+
+This scenario measures the performance of a cold load of a web page (without any cookies, storage, or anything in the cache).
+
+- [x] (some items from the previous milestones)
+- [ ] manipulating cookies
+- [ ] managing the browser cache
+- [ ] [installing bootstrap scripts](https://github.com/w3c/webdriver-bidi/blob/master/proposals/bootstrap-scripts.md#record-navigation-performance)
```

Comment:
It's not really clear to me why bootstrap scripts are really for the performance-mesaurement usecase? Is this the only thing for which they're currently used in Puppeteer? As I understand from the meeting notes, these are a relatively high priority from library authors because having scripts that you know run before any page scripts or other injected scripts unlocks the possibility to implement features that are otherwise difficult or impossible to get right in a race-free way.

Reply:
Similar to https://github.com/w3c/webdriver-bidi/pull/265#discussion_r962766000, I didn’t mean to suggest that this is the only use case.

Reply:
Are you okay with this particular e2e scenario? If not, please suggest a concrete alternate e2e scenario that depends on bootstrap scripts. I went through the examples in [your excellent explainer doc](https://github.com/w3c/webdriver-bidi/blob/master/proposals/bootstrap-scripts.md#record-navigation-performance) and the performance-related scenario intuitively felt like the most common one.

Reply:
@jgraham what do you think? On the other side we could also use an example for a MutationObserver which would not such much overlap with performance measurements via network and load events.

Reply:
Yeah, I think a scenario that involves bootstrap scripts + mutation observers to see an element as soon as it appears would be better. I think the requirement to poll for elements is one of the key weaknesses of WebDriver and one of the actual reasons that Selenium and others are interested in preload scripts.

Reply:
Alright, I'll go ahead and modify the example appropriately. Thanks.